### PR TITLE
add GetExpectedPhysicalPath in fal.

### PR DESCRIPTION
**Pull request by vwxyzh**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Common/FileAbstractLayer/EmptyFileReader.cs [hunk 1]
```diff
@@ -20,6 +20,9 @@ private EmptyFileReader()
 
         public IEnumerable<RelativePath> EnumerateFiles() => Enumerable.Empty<RelativePath>();
 
+        public IEnumerable<string> GetExpectedPhysicalPath(RelativePath file) =>
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Common/FileAbstractLayer/EmptyFileReader.cs]: naming: OutputPhysicalPaths?

**Comment by 928PJY** [comment]
>             return File.OpenRead(Environment.ExpandEnvironmentVariables(pp.PhysicalPath));

FindPhysicalPath(file).PhysicalPath? 
keep same with what you did following

---
Refers to: src/Microsoft.DocAsCode.Common/FileAbstractLayer/FileAbstractLayer.cs:60 in b478100. [](commit_id = b4781005c41ea522daec38b09927f0c6dbcf3455, deletion_comment = False)

File: src/Microsoft.DocAsCode.Plugins/RootedFileAbstractLayer.cs [hunk 2]
```diff
@@ -65,5 +65,9 @@ public void Copy(string sourceFileName, string destFileName)
 
         public string GetPhysicalPath(string file) =>
             Path.IsPathRooted(file) ? file : _impl.GetPhysicalPath(file);
+
+        public IEnumerable<string> GetExpectedPhysicalPath(string file) =>
+            Path.IsPathRooted(file) ? new[] { file } : _impl.GetExpectedPhysicalPath(file);
```
Comment (928PJY) [inline src/Microsoft.DocAsCode.Plugins/RootedFileAbstractLayer.cs]: remove new line

File: src/Microsoft.DocAsCode.Plugins/DefaultFileAbstractLayer.cs [hunk 3]
```diff
@@ -64,5 +64,9 @@ public void Copy(string sourceFileName, string destFileName)
             Path.Combine(
                 Environment.ExpandEnvironmentVariables(EnvironmentContext.OutputDirectory),
                 file);
+
+        public IEnumerable<string> GetExpectedPhysicalPath(string file) =>
+            new[] { Path.Combine(EnvironmentContext.OutputDirectory, file) };
```
Comment (928PJY) [inline src/Microsoft.DocAsCode.Plugins/DefaultFileAbstractLayer.cs]: remove newline

**Comment by vwxyzh** [comment]
>             return File.OpenRead(Environment.ExpandEnvironmentVariables(pp.PhysicalPath));

GetExpectedPhysicalPath do not expand environment variables.

---
In reply to: [331034527](https://github.com/dotnet/docfx/pull/2069#issuecomment-331034527) [](ancestors = 331034527)

---
Refers to: src/Microsoft.DocAsCode.Common/FileAbstractLayer/FileAbstractLayer.cs:60 in b478100. [](commit_id = b4781005c41ea522daec38b09927f0c6dbcf3455, deletion_comment = False)

File: src/Microsoft.DocAsCode.Common/FileAbstractLayer/EmptyFileReader.cs (hunk 1 above)
Comment (vwxyzh) [reply src/Microsoft.DocAsCode.Common/FileAbstractLayer/EmptyFileReader.cs]: It is expected input file path.

---
In reply to: [140129887](https://github.com/dotnet/docfx/pull/2069#discussion_r140129887) [](ancestors = 140129887)

**Pull request merged**